tests/migration/reset_scc_addons_values.pm

```diff
@@ -0,0 +1,47 @@
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Swap the values of SCC_ADDONS before migration.
+#
+# In migration MU jobs, SCC_ADDONS is used for both registering original
+# product and adding target product modules MU repos. But for SLE 12 and
+# SLE 15, the modules name are different. So swap SCC_ADDONS values to
+# upgraded SLE 15 modules before migration.
+# And re-create incidents repos based on the swapped SCC_ADDONS values.
+#
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
+use List::Util qw(uniq);
+use main_common qw(join_incidents_to_repo);
+
+sub run {
+
+    select_console 'root-console';
+
+    # SCC_ADDONS_2 values are upgraded 15SPx's modules, swap the SCC_ADDONS to SCC_ADDONS_2.
+    set_var('SCC_ADDONS', get_var('SCC_ADDONS_2')) if (get_var('SCC_ADDONS_2'));
+
+    # Recreate the incidents repos based on the swapped SCC_ADDONS values.
+    my %incidents;
+    $incidents{OS} = get_var('OS_TEST_REPOS', '');
```

Comment:
this could go in single line and apparently we don't use `OS_TEST_REPOS` so it could be initialize it to empty. But I would recommend you something else, have you ever use `map` in Perl?
You can iterate @addons and then map the hash ref %incidents in a single line without any loop.
https://perldoc.perl.org/functions/map

BUT actually a 3rd option is not to use the existing code from main.pm, we just need a variable with all the repos concatenated, why we don't do that in a loop concatenating each `*_TEST_REPOS` that we have based in our know addons? would be more simple I guess.